### PR TITLE
Fix error handling in Rest API calls and refactor console logs

### LINKED ISSUE #91: Fix API request error handling in Dagre ScriptPage component
After ticket #86, the behavior in the dagre file is incorrect.

The current implementation checks the status explicitly, but since Axios already handles the status checks, you should use the .catch() method to handle any errors in the promise.

For example this:

```javascript
 Rest.deleteScriptNode(filepath, ele.data("id")).then((response) => {
              this.setState({ isLoaded: true });
              if (response.status === 204) {
                ele.remove();
              } else {
                console.log("Node can not be deleted.");
                this.setState({ errorMessage: "Node can not be deleted." });
              }
            });
  ```

should be rewritten to

```javascript
Rest.deleteScriptNode(filepath, ele.data("id"))
  .then((response) => {
    this.setState({ isLoaded: true });
    ele.remove();
  })
  .catch((error) => {
    console.log("Node cannot be deleted.");
    this.setState({ errorMessage: "Node cannot be deleted." });
  });
  ```

### PR DESCRIPTION
**Pull request by palagdan**
Resolve #91

### CODE REVIEW

File: src/components/modal/ScriptOntologyModal.jsx [hunk 1]
```diff
@@ -30,37 +30,35 @@ class ScriptOntologyModal extends React.Component {
...
+    try {
+      console.log(this.state.scriptPath + " #### " + ontologyURI);
+      await Rest.removeScriptOwnOntology(this.state.scriptPath, ontologyURI);
+      const response = await Rest.getScriptOwnOntology(this.state.scriptPath);
+      this.setState({
+        scriptPath: this.state.scriptPath,
+        ontologies: response,
+      });
+    } catch (error) {
+      alert(`An error occurred while removing the ontology: ${ontologyURI}. ${error.message || error}`);
```
Comment (blcham) [inline src/components/modal/ScriptOntologyModal.jsx:43]: This looks like inconsistent with the rest of what you wrote ... you always logged using
alert() --- which was less specific
console.log() -- which was more specific

Here, it is the opposite way around ... 

Moreover, here you use console.error and elsewhere you use console.log ... 


Is it the way you wanted ?

**Review by blcham — CHANGES_REQUESTED** [review]
LGTM, just make sure that you are consistent when logging different errors. See my  comment above.

**Comment by blcham** [comment]
:eyes:

File: src/components/modal/ScriptOntologyModal.jsx (hunk 1 above)
Comment (palagdan) [reply src/components/modal/ScriptOntologyModal.jsx]: Yes, console.log will provide more detailed information about the error, while alert will only provide basic UI information.

About logs, I think a better solution would be to use console.error for errors, as it helps differentiate between logs related to states and actual error logs.

File: src/components/modal/ScriptOntologyModal.jsx [hunk 2]
```diff
@@ -30,37 +30,35 @@ class ScriptOntologyModal extends React.Component {
...
+      console.log(this.state.scriptPath + " #### " + ontologyURI);
+      await Rest.removeScriptOwnOntology(this.state.scriptPath, ontologyURI);
+      const response = await Rest.getScriptOwnOntology(this.state.scriptPath);
+      this.setState({
+        scriptPath: this.state.scriptPath,
+        ontologies: response,
+      });
+    } catch (error) {
+      alert("An error occurred while removing the ontology.");
+      console.error(`Error removing ontology ${ontologyURI}: ${error}`);
```
[the remark this answers is not in the document]
Comment (palagdan) [reply src/components/modal/ScriptOntologyModal.jsx:44]: I will do it in this PR

**Review by blcham — APPROVED** [review]
LGTM

**Pull request merged**